### PR TITLE
(No)Ethersjs switch

### PR DESCRIPTION
**Pull request by szerintedmi**
Attempt to switch to etherjs lib but rolled back at the end - it introduces more issues than it solves. We should revisit if we still need it for the event handling.
But fixed a ton of things on the way, eg:

- cleaned up values in  store
- bignumber calculations rationalising (we might need to revisit, some places we might need to re-introduce bignumber calculations, [see more here](https://docs.ethers.io/ethers.js/html/notes.html#why-can-t-i-just-use-numbers))
- transaction error and success handling refactor
- loads of small fixes

This is a commit I'm not proud of because of size, apologies. I understand if you can't review it properly.

### CODE REVIEW

**Comment by szerintedmi** [comment]
Deploy preview for *augmint* processing.

Built with commit 7ca73a8b5949b133bfe809ff373503ac4ef4abf0

https://app.netlify.com/sites/augmint/deploys/5a8dccdea6188f1d6a731cf2

File: src/components/MsgPanels.js [hunk 1]
```diff
@@ -140,11 +141,20 @@ export function ConnectionStatus(props) {
...
+    const { header = "Error details:", style = { fontSize: "0.8em", overflow: "auto" }, details } = props;
+
+    return !details ? null : (
         <div>
             <p>{header}</p>
-            <pre style={style}>{props.children}</pre>
+
+            {details && <pre style={style}>{details.message ? details.message : details}</pre>}
+
+            {/* {details.stack (
```
Comment (gyurka00) [inline src/components/MsgPanels.js]: why are there commented lines?

Comment (szerintedmi) [reply src/components/MsgPanels.js]: stack was quite unhelpful to display. content of it is in the message anyway (eg. try to reject tx signature in metamask to test)
we will need to tweak this error handling in the future, hide the sacry details by default and handle basic scenarios (e.g signature rejection, metamask is locked etc.). I think we've a trello card for it.

File: src/modules/reducers/web3Connect.js [hunk 2]
```diff
@@ -85,6 +85,7 @@ export const setupWeb3 = () => {
                 console.debug(
                     "No web3 detected. Falling back to http://localhost:8545. You should remove this fallback when you deploy live, as it's inherently insecure. Consider switching to Metamask for development. More info here: http://truffleframework.com/tutorials/truffle-and-metamask"
                 );
+                //const web3Provider = new Web3.providers.WebsocketProvider("ws://localhost:8545");
```
Comment (gyurka00) [inline src/modules/reducers/web3Connect.js]: If we do not need this commented line please delete it.

Comment (szerintedmi) [reply src/modules/reducers/web3Connect.js]: done :)

**Pull request merged**